### PR TITLE
[tensorexpr] support for fusing autocasting ops

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72478

aten::_autocast_to_reduced_precision and `aten::_autocast_to_full_precision are essentially just aten::to operations, so they can be fused the same way aten::to is fused.

Differential Revision: [D34057522](https://our.internmc.facebook.com/intern/diff/D34057522)

### CODE REVIEW

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

**Comment by davidberard98** [comment]
RUN_TORCHBENCH: ALL

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 1]
```diff
@@ -981,16 +996,23 @@ class TensorExprFuser {
           return false;
         }
       }
-      // cant support non-constant pin_memory or pin_memory = True
```
Comment (eellison) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp:984]: Let's add the autocast nodes here: https://github.com/pytorch/pytorch/blob/95e6722e852d9d13f528945648b5ac6f086983c1/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L622

**Review by eellison — APPROVED** [review]
LGTM with one change

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057522).

**Pull request closed without merging**